### PR TITLE
pacific: qa/suites/upgrade/pacific-p2p: run librbd python API tests from pacific tip

### PR DESCRIPTION
**Pull request by yuriw**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by yuriw** [comment]
https://pulpito.ceph.com/yuriw-2024-02-01_23:40:28-upgrade:pacific-p2p-pacific-release-distro-default-smithi/

File: qa/suites/upgrade/pacific-p2p/pacific-p2p-stress-split/6-final-workload/rbd-python.yaml [hunk 1]
```diff
@@ -6,5 +6,5 @@ tasks:
     tag: v16.2.7
     clients:
       client.0:
-        - rbd/test_librbd_python.sh
+        - rbd/test_librbd_python.sh -k "not test_diff_iterate"
```
Comment (idryomov) [inline qa/suites/upgrade/pacific-p2p/pacific-p2p-stress-split/6-final-workload/rbd-python.yaml:9]: This doesn't work:
- `test_librbd_python.sh` doesn't propagate its arguments to `nose`
- Separately, I don't think `nose` has `-k` option.  Can you point at where it's documented?

**Review by idryomov — COMMENTED** [review]
The commit message is empty.  When disabling/skipping tests like this, at least a minimal explanation is expected.

**Comment by yuriw** [comment]
removed 6-final-workload/rbd-python.yaml

https://pulpito.ceph.com/yuriw-2024-02-02_20:21:37-upgrade:pacific-p2p-pacific-release-distro-default-smithi/

**Comment by idryomov** [comment]
So what this job is doing wrt. librbd:

1. Install librbd from v16.2.7

```
2024-02-02T02:39:17.247 INFO:teuthology.orchestra.run.smithi121.stdout:Get:2 https://chacra.ceph.com/r/ceph/pacific/dd0603118f56ab514f133c8d2e3adfc983942503/ubuntu/focal/flavors/default focal/main amd64 librbd1 amd64 16.2.7-1focal [3,319 kB]
2024-02-02T02:39:20.402 INFO:teuthology.orchestra.run.smithi121.stdout:Preparing to unpack .../02-librbd1_16.2.7-1focal_amd64.deb ...
2024-02-02T02:39:20.528 INFO:teuthology.orchestra.run.smithi121.stdout:Unpacking librbd1 (16.2.7-1focal) over (15.2.17-0ubuntu0.20.04.6) ...
2024-02-02T02:40:35.933 INFO:teuthology.orchestra.run.smithi121.stdout:Setting up librbd1 (16.2.7-1focal) ...
```

2. Upgrade librbd to latest pacific

```
2024-02-02T04:14:10.638 INFO:teuthology.orchestra.run.smithi121.stdout:Unpacking librbd1 (16.2.14-510-g88fb4c6a-1focal) over (16.2.7-1focal) ...
```

3. Run `test_librbd_python.sh` tests from v16.2.7 against the upgraded librbd

```
2024-02-02T04:18:17.906 INFO:teuthology.run_tasks:Running task workunit...
2024-02-02T04:18:17.917 INFO:tasks.workunit:Pulling workunits from ref v16.2.7
...
2024-02-02T04:18:17.940 DEBUG:teuthology.orchestra.run.smithi121:> rm -rf /home/ubuntu/cephtest/clone.client.0 && git clone https://github.com/ceph/ceph.git /home/ubuntu/cephtest/clone.client.0 && cd /home/ubuntu/cephtest/clone.client.0 && git checkout v16.2.7
2024-02-02T04:18:17.991 INFO:tasks.workunit.client.0.smithi121.stderr:Cloning into '/home/ubuntu/cephtest/clone.client.0'...                                                                  
2024-02-02T04:19:00.630 INFO:tasks.workunit.client.0.smithi121.stderr:Note: switching to 'v16.2.7'.
```

This order is guaranteed because (2) is `5-finish-upgrade.yaml`, while (3) is under `6-final-workload`.

We can't account for such version mismatches when writing tests.  However, instead of removing the job altogether, I'd suggest replacing `tag: v16.2.7` with `branch: pacific` there.

**Comment by yuriw** [comment]
Trying @idryomov 's suggestion
https://pulpito.ceph.com/yuriw-2024-02-08_17:51:40-upgrade:pacific-p2p-pacific-release-distro-default-smithi/

**Comment by idryomov** [comment]
> Trying @idryomov 's suggestion https://pulpito.ceph.com/yuriw-2024-02-08_17:51:40-upgrade:pacific-p2p-pacific-release-distro-default-smithi/

It worked!  Please updated the commit title and commit message to:
```
qa/suites/upgrade/pacific-p2p: run librbd python API tests from pacific tip

This jobs installs librbd from v16.2.7, upgrades librbd to latest
pacific and then runs librbd python API tests from v16.2.7 against the
upgraded librbd.  This isn't expected to work in the general case
and is currently failing in TestImage.test_diff_iterate test because
it was adjusted to do the right thing in commit f8ced6d1fe66
("test/pybind/rbd: don't ignore from_snapshot in check_diff()").

Co-authored-by: Ilya Dryomov <idryomov@gmail.com>
Signed-off-by: Yuri Weinstein <yweinste@redhat.com>
```

**Comment by yuriw** [comment]
> > Trying @idryomov 's suggestion https://pulpito.ceph.com/yuriw-2024-02-08_17:51:40-upgrade:pacific-p2p-pacific-release-distro-default-smithi/
> 
> It worked! Please update the commit title and commit message to:
> 
> ```
> qa/suites/upgrade/pacific-p2p: run librbd python API tests from pacific tip
> 
> This job installs librbd from v16.2.7, upgrades librbd to latest
> pacific and then runs librbd python API tests from v16.2.7 against the
> upgraded librbd.  This isn't expected to work in the general case
> and is currently failing in TestImage.test_diff_iterate test because
> it was adjusted to do the right thing in commit f8ced6d1fe66
> ("test/pybind/rbd: don't ignore from_snapshot in check_diff()").
> 
> Co-authored-by: Ilya Dryomov <idryomov@gmail.com>
> Signed-off-by: Yuri Weinstein <yweinste@redhat.com>
> ```

Done

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
jenkins test api

**Pull request merged**